File: multi_agent/projects/im-desktop/services/contextAwareReplyService.ts
import axios, { AxiosInstance } from 'axios';
import {
  ContextAwareReply,
  GenerateReplyRequest,
  GenerateReplyResponse,
  FeedbackRequest,
  SearchReplyRequest,
  PageResponse,
  StatisticsResponse,
  HealthResponse,
  ALL_STATUSES
} from './contextAwareReplyTypes';

/**
 * 上下文感知智能回复生成器 API 服务
 */
class ContextAwareReplyService {
  private api: AxiosInstance;
  private baseURL: string;

  constructor(baseURL: string = '/api/v1') {
    this.baseURL = baseURL;
    this.api = axios.create({
      baseURL,
      timeout: 30000,
      headers: {
        'Content-Type': 'application/json'
      }
    });

    // 请求拦截器
    this.api.interceptors.request.use(
      (config) => {
        // 添加认证token
        const token = localStorage.getItem('token');
        if (token) {
          config.headers.Authorization = `Bearer ${token}`;
        }
        return config;
      },
      (error) => Promise.reject(error)
    );

    // 响应拦截器
    this.api.interceptors.response.use(
      (response) => response.data,
      (error) => {
        console.error('API Error:', error.response?.data || error.message);
        return Promise.reject(error);
      }
    );
  }

  // ==================== 基础 CRUD 操作 ====================

  /**
   * 创建智能回复记录
   */
  async createReply(reply: Partial<ContextAwareReply>): Promise<ContextAwareReply> {
    const response = await this.api.post('/context-aware-reply', reply);
    return response as ContextAwareReply;
  }

  /**
   * 批量创建回复记录
   */
  async createReplies(replies: Partial<ContextAwareReply>[]): Promise<ContextAwareReply[]> {
    const response = await this.api.post('/context-aware-reply/batch', replies);
    return response as ContextAwareReply[];
  }

  /**
   * 获取回复详情
   */
  async getReplyById(id: number): Promise<ContextAwareReply> {
    const response = await this.api.get(`/context-aware-reply/${id}`);
    return response as ContextAwareReply;
  }

  /**
   * 更新回复记录
   */
  async updateReply(id: number, updateData: Partial<ContextAwareReply>): Promise<ContextAwareReply> {
    const response = await this.api.put(`/context-aware-reply/${id}`, updateData);
    return response as ContextAwareReply;
  }

  /**
   * 删除回复记录
   */
  async deleteReply(id: number): Promise<void> {
    await this.api.delete(`/context-aware-reply/${id}`);
  }

  /**
   * 批量删除回复记录
   */
  async deleteReplies(ids: number[]): Promise<void> {
    await this.api.delete('/context-aware-reply/batch', { data: ids });
  }

  // ==================== 查询操作 ====================

  /**
   * 获取用户的回复记录
   */
  async getRepliesByUser(userId: string): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/user/${userId}`);
    return response as ContextAwareReply[];
  }

  /**
   * 分页获取用户的回复记录
   */
  async getRepliesByUserPaged(
    userId: string,
    page: number = 0,
    size: number = 20
  ): Promise<PageResponse<ContextAwareReply>> {
    const response = await this.api.get(`/context-aware-reply/user/${userId}/paged`, {
      params: { page, size }
    });
    return response as PageResponse<ContextAwareReply>;
  }

  /**
   * 获取会话的回复记录
   */
  async getRepliesBySession(sessionId: string): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/session/${sessionId}`);
    return response as ContextAwareReply[];
  }

  /**
   * 获取用户和会话的回复记录
   */
  async getRepliesByUserAndSession(
    userId: string,
    sessionId: string
  ): Promise<ContextAwareReply[]> {
    const response = await this.api.get(
      `/context-aware-reply/user/${userId}/session/${sessionId}`
    );
    return response as ContextAwareReply[];
  }

  /**
   * 获取触发消息的回复记录
   */
  async getReplyByTriggerMessage(messageId: string): Promise<ContextAwareReply | null> {
    try {
      const response = await this.api.get(`/context-aware-reply/trigger-message/${messageId}`);
      return response as ContextAwareReply;
    } catch (error: any) {
      if (error.response?.status === 404) {
        return null;
      }
      throw error;
    }
  }

  /**
   * 获取指定状态的回复记录
   */
  async getRepliesByStatus(status: string): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/status/${status}`);
    return response as ContextAwareReply[];
  }

  /**
   * 获取已使用的回复记录
   */
  async getUsedReplies(): Promise<ContextAwareReply[]> {
    const response = await this.api.get('/context-aware-reply/used');
    return response as ContextAwareReply[];
  }

  /**
   * 获取高质量的回复记录
   */
  async getHighQualityReplies(): Promise<ContextAwareReply[]> {
    const response = await this.api.get('/context-aware-reply/high-quality');
    return response as ContextAwareReply[];
  }

  /**
   * 获取高置信度的回复记录
   */
  async getHighConfidenceReplies(): Promise<ContextAwareReply[]> {
    const response = await this.api.get('/context-aware-reply/high-confidence');
    return response as ContextAwareReply[];
  }

  // ==================== 意图相关查询 ====================

  /**
   * 获取指定意图的回复记录
   */
  async getRepliesByIntent(intent: string): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/intent/${intent}`);
    return response as ContextAwareReply[];
  }

  /**
   * 获取用户指定意图的回复记录
   */
  async getRepliesByUserAndIntent(
    userId: string,
    intent: string
  ): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/user/${userId}/intent/${intent}`);
    return response as ContextAwareReply[];
  }

  /**
   * 获取用户最常用的意图
   */
  async getUserTopIntents(userId: string): Promise<Record<string, number>> {
    const response = await this.api.get(`/context-aware-reply/user/${userId}/top-intents`);
    return response as Record<string, number>;
  }

  // ==================== 语言风格相关查询 ====================

  /**
   * 获取指定语言风格的回复记录
   */
  async getRepliesByLanguageStyle(style: string): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/language-style/${style}`);
    return response as ContextAwareReply[];
  }

  /**
   * 获取用户指定语言风格的回复记录
   */
  async getRepliesByUserAndLanguageStyle(
    userId: string,
    style: string
  ): Promise<ContextAwareReply[]> {
    const response = await this.api.get(
      `/context-aware-reply/user/${userId}/language-style/${style}`
    );
    return response as ContextAwareReply[];
  }

  /**
   * 获取用户最常用的语言风格
   */
  async getUserTopLanguageStyles(userId: string): Promise<Record<string, number>> {
    const response = await this.api.get(`/context-aware-reply/user/${userId}/top-language-styles`);
    return response as Record<string, number>;
  }

  // ==================== 时间范围查询 ====================

  /**
   * 获取时间范围内的回复记录
   */
  async getRepliesByDateRange(
    start: string,
    end: string
  ): Promise<ContextAwareReply[]> {
    const response = await this.api.get('/context-aware-reply/date-range', {
      params: { start, end }
    });
    return response as ContextAwareReply[];
  }

  /**
   * 获取用户时间范围内的回复记录
   */
  async getRepliesByUserAndDateRange(
    userId: string,
    start: string,
    end: string
  ): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/user/${userId}/date-range`, {
      params: { start, end }
    });
    return response as ContextAwareReply[];
  }

  // ==================== 统计操作 ====================

  /**
   * 统计用户回复记录数量
   */
  async countRepliesByUser(userId: string): Promise<number> {
    const response = await this.api.get(`/context-aware-reply/count/user/${userId}`);
    return response as number;
  }

  /**
   * 统计会话回复记录数量
   */
  async countRepliesBySession(sessionId: string): Promise<number> {
    const response = await this.api.get(`/context-aware-reply/count/session/${sessionId}`);
    return response as number;
  }

  /**
   * 统计指定状态回复记录数量
   */
  async countRepliesByStatus(status: string): Promise<number> {
    const response = await this.api.get(`/context-aware-reply/count/status/${status}`);
    return response as number;
  }

  /**
   * 统计已使用回复记录数量
   */
  async countUsedReplies(): Promise<number> {
    const response = await this.api.get('/context-aware-reply/count/used');
    return response as number;
  }

  /**
   * 统计高质量回复记录数量
   */
  async countHighQualityReplies(): Promise<number> {
    const response = await this.api.get('/context-aware-reply/count/high-quality');
    return response as number;
  }

  /**
   * 获取意图分布统计
   */
  async getIntentDistribution(): Promise<Record<string, number>> {
    const response = await this.api.get('/context-aware-reply/stats/intent-distribution');
    return response as Record<string, number>;
  }

  /**
   * 获取语言风格分布统计
   */
  async getLanguageStyleDistribution(): Promise<Record<string, number>> {
    const response = await this.api.get('/context-aware-reply/stats/language-style-distribution');
    return response as Record<string, number>;
  }

  /**
   * 获取平均反馈评分
   */
  async getAverageFeedbackScore(): Promise<number> {
    const response = await this.api.get('/context-aware-reply/stats/average-feedback-score');
    return response as number;
  }

  /**
   * 获取平均生成时间
   */
  async getAverageGenerationTime(): Promise<number> {
    const response = await this.api.get('/context-aware-reply/stats/average-generation-time');
    return response as number;
  }

  /**
   * 获取完整统计信息
   */
  async getStatistics(): Promise<StatisticsResponse> {
    const [
      totalReplies,
      usedReplies,
      highQualityReplies,
      averageFeedbackScore,
      averageGenerationTimeMs,
      intentDistribution,
      languageStyleDistribution
    ] = await Promise.all([
      this.countRepliesByStatus(ALL_STATUSES[0]), // 简单处理，实际需要调整
      this.countUsedReplies(),
      this.countHighQualityReplies(),
      this.getAverageFeedbackScore(),
      this.getAverageGenerationTime(),
      this.getIntentDistribution(),
      this.getLanguageStyleDistribution()
    ]);

    return {
      totalReplies,
      totalUsers: 0, // 需要后端支持
      usedReplies,
      highQualityReplies,
      averageFeedbackScore,
      averageGenerationTimeMs,
      intentDistribution,
      languageStyleDistribution,
      statusDistribution: {}
    };
  }

  // ==================== 高级操作 ====================

  /**
   * 搜索回复记录
   */
  async searchReplies(keyword: string): Promise<ContextAwareReply[]> {
    const response = await this.api.get('/context-aware-reply/search', {
      params: { keyword }
    });
    return response as ContextAwareReply[];
  }

  /**
   * 标记回复为已使用
   */
  async markAsUsed(id: number): Promise<void> {
    await this.api.post(`/context-aware-reply/${id}/mark-used`);
  }

  /**
   * 批量标记为已使用
   */
  async markMultipleAsUsed(ids: number[]): Promise<void> {
    await this.api.post('/context-aware-reply/batch/mark-used', ids);
  }

  /**
   * 提交用户反馈
   */
  async submitFeedback(id: number, score: number, comment?: string): Promise<void> {
    await this.api.post(`/context-aware-reply/${id}/feedback`, null, {
      params: { score, comment }
    });
  }

  /**
   * 获取最近N条用户回复
   */
  async getRecentRepliesByUser(userId: string, limit: number = 10): Promise<ContextAwareReply[]> {
    const response = await this.api.get(`/context-aware-reply/user/${userId}/recent`, {
      params: { limit }
    });
    return response as ContextAwareReply[];
  }

  /**
   * 清理过期回复
   */
  async cleanupExpiredReplies(): Promise<number> {
    const response = await this.api.post('/context-aware-reply/cleanup/expired');
    return response as number;
  }

  /**
   * 清理低质量回复
   */
  async cleanupLowQualityReplies(): Promise<number> {
    const response = await this.api.post('/context-aware-reply/cleanup/low-quality');
    return response as number;
  }

  // ==================== 内容获取 ====================

  /**
   * 获取回复候选列表
   */
  async getReplyCandidates(id: number): Promise<string[]> {
    const response = await this.api.get(`/context-aware-reply/${id}/candidates`);
    return response as string[];
  }

  /**
   * 获取推荐的表情符号列表
   */
  async getRecommendedEmojis(id: number): Promise<string[]> {
    const response = await this.api.get(`/context-aware-reply/${id}/emojis`);
    return response as string[];
  }

  /**
   * 获取个性化特征
   */
  async getPersonalizationFeatures(id: number): Promise<Record<string, any>> {
    const response = await this.api.get(`/context-aware-reply/${id}/personalization`);
    return response as Record<string, any>;
  }

  /**
   * 获取生成选项
   */
  async getGenerationOptions(id: number): Promise<Record<string, any>> {
    const response = await this.api.get(`/context-aware-reply/${id}/options`);
    return response as Record<string, any>;
  }

  // ==================== 智能生成 ====================

  /**
   * 生成智能回复
   */
  async generateReply(request: GenerateReplyRequest): Promise<ContextAwareReply> {
    const response = await this.api.post('/context-aware-reply/generate', null, {
      params: {
        userId: request.userId,
        sessionId: request.sessionId,
        triggerMessageContent: request.triggerMessageContent
      }
    });
    return response as ContextAwareReply;
  }

  // ==================== 健康检查 ====================

  /**
   * 健康检查
   */
  async healthCheck(): Promise<HealthResponse> {
    const response = await this.api.get('/context-aware-reply/health');
    return response as HealthResponse;
  }
}

// 导出单例
export const contextAwareReplyService = new ContextAwareReplyService();
export default contextAwareReplyService;

File: im-mobile/lib/services/context_aware_reply_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

/// 意图枚举
enum ReplyIntent {
  BUSINESS,
  PERSONAL,
  SOCIAL,
  QUESTION,
  GREETING,
  APPRECIATION,
  EMOTIONAL,
}

/// 语言风格枚举
enum LanguageStyle {
  FORMAL,
  CASUAL,
  FRIENDLY,
  PROFESSIONAL,
}

/// 回复长度枚举
enum ReplyLength {
  SHORT,
  MEDIUM,
  LONG,
}

/// 状态枚举
enum ReplyStatus {
  GENERATED,
  SELECTED,
  REJECTED,
  EXPIRED,
}

/// 上下文感知智能回复生成器模型
class ContextAwareReply {
  final int id;
  final String userId;
  final String? sessionId;
  final String? triggerMessageId;
  final String? triggerMessageContent;
  final String? contextSummary;
  final String? detectedIntent;
  final double? intentConfidence;
  final List<String>? replyCandidates;
  final String? selectedReply;
  final List<String>? recommendedEmojis;
  final String? languageStyle;
  final String? replyLength;
  final String? sensitivityCheckResult;
  final bool? sensitivityPassed;
  final Map<String, dynamic>? personalizationFeatures;
  final int? userFeedbackScore;
  final String? userFeedbackComment;
  final bool? used;
  final int? generationTimeMs;
  final String? modelVersion;
  final Map<String, dynamic>? generationOptions;
  final String status;
  final DateTime? expiresAt;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final String? indexKey;

  ContextAwareReply({
    required this.id,
    required this.userId,
    this.sessionId,
    this.triggerMessageId,
    this.triggerMessageContent,
    this.contextSummary,
    this.detectedIntent,
    this.intentConfidence,
    this.replyCandidates,
    this.selectedReply,
    this.recommendedEmojis,
    this.languageStyle,
    this.replyLength,
    this.sensitivityCheckResult,
    this.sensitivityPassed,
    this.personalizationFeatures,
    this.userFeedbackScore,
    this.userFeedbackComment,
    this.used,
    this.generationTimeMs,
    this.modelVersion,
    this.generationOptions,
    required this.status,
    this.expiresAt,
    this.createdAt,
    this.updatedAt,
    this.indexKey,
  });

  factory ContextAwareReply.fromJson(Map<String, dynamic> json) {
    return ContextAwareReply(
      id: json['id'] ?? 0,
      userId: json['userId'] ?? '',
      sessionId: json['sessionId'],
      triggerMessageId: json['triggerMessageId'],
      triggerMessageContent: json['triggerMessageContent'],
      contextSummary: json['contextSummary'],
      detectedIntent: json['detectedIntent'],
      intentConfidence: json['intentConfidence']?.toDouble(),
      replyCandidates: json['replyCandidates'] != null
          ? List<String>.from(json['replyCandidates'])
          : null,
      selectedReply: json['selectedReply'],
      recommendedEmojis: json['recommendedEmojis'] != null
          ? List<String>.from(json['recommendedEmojis'])
          : null,
      languageStyle: json['languageStyle'],
      replyLength: json['replyLength'],
      sensitivityCheckResult: json['sensitivityCheckResult'],
      sensitivityPassed: json['sensitivityPassed'],
      personalizationFeatures: json['personalizationFeatures'] != null
          ? Map<String, dynamic>.from(json['personalizationFeatures'])
          : null,
      userFeedbackScore: json['userFeedbackScore'],
      userFeedbackComment: json['userFeedbackComment'],
      used: json['used'],
      generationTimeMs: json['generationTimeMs'],
      modelVersion: json['modelVersion'],
      generationOptions: json['generationOptions'] != null
          ? Map<String, dynamic>.from(json['generationOptions'])
          : null,
      status: json['status'] ?? 'GENERATED',
      expiresAt: json['expiresAt'] != null
          ? DateTime.parse(json['expiresAt'])
          : null,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'])
          : null,
      indexKey: json['indexKey'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'sessionId': sessionId,
      'triggerMessageId': triggerMessageId,
      'triggerMessageContent': triggerMessageContent,
      'contextSummary': contextSummary,
      'detectedIntent': detectedIntent,
      'intentConfidence': intentConfidence,
      'replyCandidates': replyCandidates,
      'selectedReply': selectedReply,
      'recommendedEmojis': recommendedEmojis,
      'languageStyle': languageStyle,
      'replyLength': replyLength,
      'sensitivityCheckResult': sensitivityCheckResult,
      'sensitivityPassed': sensitivityPassed,
      'personalizationFeatures': personalizationFeatures,
      'userFeedbackScore': userFeedbackScore,
      'userFeedbackComment': userFeedbackComment,
      'used': used,
      'generationTimeMs': generationTimeMs,
      'modelVersion': modelVersion,
      'generationOptions': generationOptions,
      'status': status,
      'expiresAt': expiresAt?.toIso8601String(),
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
      'indexKey': indexKey,
    };
  }

  bool get isHighQuality => (userFeedbackScore ?? 0) >= 4;
  bool get isExpired => expiresAt != null && DateTime.now().isAfter(expiresAt!);
  bool get isUsable =>
      status == 'GENERATED' && !isExpired && (sensitivityPassed ?? true);
}

/// 生成回复请求
class GenerateReplyRequest {
  final String userId;
  final String? sessionId;
  final String triggerMessageContent;
  final Map<String, dynamic>? context;
  final String? languageStyle;
  final String? replyLength;
  final Map<String, dynamic>? generationOptions;

  GenerateReplyRequest({
    required this.userId,
    this.sessionId,
    required this.triggerMessageContent,
    this.context,
    this.languageStyle,
    this.replyLength,
    this.generationOptions,
  });

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'sessionId': sessionId,
      'triggerMessageContent': triggerMessageContent,
      'context': context,
      'languageStyle': languageStyle,
      'replyLength': replyLength,
      'generationOptions': generationOptions,
    };
  }
}

/// 反馈请求
class FeedbackRequest {
  final int score;
  final String? comment;

  FeedbackRequest({
    required this.score,
    this.comment,
  });

  Map<String, dynamic> toJson() {
    return {
      'score': score,
      'comment': comment,
    };
  }

  bool get isValid => score >= 1 && score <= 5;
}

/// 上下文感知智能回复生成器 API 服务
class ContextAwareReplyApiService {
  final String baseUrl;
  final http.Client _client;
  String? _authToken;

  ContextAwareReplyApiService({
    this.baseUrl = '/api/v1',
    http.Client? client,
  })  : _client = client ?? http.Client();

  void setAuthToken(String token) {
    _authToken = token;
  }

  Map<String, String> get _headers => {
        'Content-Type': 'application/json',
        if (_authToken != null) 'Authorization': 'Bearer $_authToken',
      };

  // ==================== 基础 CRUD 操作 ====================

  /// 创建智能回复记录
  Future<ContextAwareReply> createReply(ContextAwareReply reply) async {
    final response = await _client.post(
      Uri.parse('$baseUrl/context-aware-reply'),
      headers: _headers,
      body: jsonEncode(reply.toJson()),
    );
    _checkError(response);
    return ContextAwareReply.fromJson(jsonDecode(response.body));
  }

  /// 获取回复详情
  Future<ContextAwareReply> getReplyById(int id) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/$id'),
      headers: _headers,
    );
    _checkError(response);
    return ContextAwareReply.fromJson(jsonDecode(response.body));
  }

  /// 更新回复记录
  Future<ContextAwareReply> updateReply(
      int id, Map<String, dynamic> updateData) async {
    final response = await _client.put(
      Uri.parse('$baseUrl/context-aware-reply/$id'),
      headers: _headers,
      body: jsonEncode(updateData),
    );
    _checkError(response);
    return ContextAwareReply.fromJson(jsonDecode(response.body));
  }

  /// 删除回复记录
  Future<void> deleteReply(int id) async {
    final response = await _client.delete(
      Uri.parse('$baseUrl/context-aware-reply/$id'),
      headers: _headers,
    );
    _checkError(response);
  }

  // ==================== 查询操作 ====================

  /// 获取用户的回复记录
  Future<List<ContextAwareReply>> getRepliesByUser(String userId) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/user/$userId'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  /// 获取会话的回复记录
  Future<List<ContextAwareReply>> getRepliesBySession(String sessionId) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/session/$sessionId'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  /// 获取用户和会话的回复记录
  Future<List<ContextAwareReply>> getRepliesByUserAndSession(
      String userId, String sessionId) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/user/$userId/session/$sessionId'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  /// 获取指定状态的回复记录
  Future<List<ContextAwareReply>> getRepliesByStatus(String status) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/status/$status'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  /// 获取已使用的回复记录
  Future<List<ContextAwareReply>> getUsedReplies() async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/used'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  /// 获取高质量的回复记录
  Future<List<ContextAwareReply>> getHighQualityReplies() async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/high-quality'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  // ==================== 意图相关查询 ====================

  /// 获取指定意图的回复记录
  Future<List<ContextAwareReply>> getRepliesByIntent(String intent) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/intent/$intent'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  /// 获取用户指定意图的回复记录
  Future<List<ContextAwareReply>> getRepliesByUserAndIntent(
      String userId, String intent) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/user/$userId/intent/$intent'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  /// 获取用户最常用的意图
  Future<Map<String, int>> getUserTopIntents(String userId) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/user/$userId/top-intents'),
      headers: _headers,
    );
    _checkError(response);
    final Map<String, dynamic> jsonMap = jsonDecode(response.body);
    return jsonMap.map((key, value) => MapEntry(key, value as int));
  }

  // ==================== 统计操作 ====================

  /// 统计用户回复记录数量
  Future<int> countRepliesByUser(String userId) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/count/user/$userId'),
      headers: _headers,
    );
    _checkError(response);
    return jsonDecode(response.body) as int;
  }

  /// 统计已使用回复记录数量
  Future<int> countUsedReplies() async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/count/used'),
      headers: _headers,
    );
    _checkError(response);
    return jsonDecode(response.body) as int;
  }

  /// 统计高质量回复记录数量
  Future<int> countHighQualityReplies() async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/count/high-quality'),
      headers: _headers,
    );
    _checkError(response);
    return jsonDecode(response.body) as int;
  }

  /// 获取平均反馈评分
  Future<double> getAverageFeedbackScore() async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/stats/average-feedback-score'),
      headers: _headers,
    );
    _checkError(response);
    return (jsonDecode(response.body) as num).toDouble();
  }

  /// 获取平均生成时间
  Future<double> getAverageGenerationTime() async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/stats/average-generation-time'),
      headers: _headers,
    );
    _checkError(response);
    return (jsonDecode(response.body) as num).toDouble();
  }

  // ==================== 高级操作 ====================

  /// 标记回复为已使用
  Future<void> markAsUsed(int id) async {
    final response = await _client.post(
      Uri.parse('$baseUrl/context-aware-reply/$id/mark-used'),
      headers: _headers,
    );
    _checkError(response);
  }

  /// 提交用户反馈
  Future<void> submitFeedback(int id, int score, String? comment) async {
    final response = await _client.post(
      Uri.parse('$baseUrl/context-aware-reply/$id/feedback'),
      headers: _headers,
      body: jsonEncode({'score': score, 'comment': comment}),
    );
    _checkError(response);
  }

  /// 获取最近 N 条用户回复
  Future<List<ContextAwareReply>> getRecentRepliesByUser(
      String userId, int limit) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/user/$userId/recent?limit=$limit'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((json) => ContextAwareReply.fromJson(json)).toList();
  }

  // ==================== 内容获取 ====================

  /// 获取回复候选列表
  Future<List<String>> getReplyCandidates(int id) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/$id/candidates'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((item) => item as String).toList();
  }

  /// 获取推荐的表情符号列表
  Future<List<String>> getRecommendedEmojis(int id) async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/$id/emojis'),
      headers: _headers,
    );
    _checkError(response);
    final List<dynamic> jsonList = jsonDecode(response.body);
    return jsonList.map((item) => item as String).toList();
  }

  // ==================== 智能生成 ====================

  /// 生成智能回复
  Future<ContextAwareReply> generateReply(GenerateReplyRequest request) async {
    final response = await _client.post(
      Uri.parse('$baseUrl/context-aware-reply/generate'),
      headers: _headers,
      body: jsonEncode({
        'userId': request.userId,
        'sessionId': request.sessionId,
        'triggerMessageContent': request.triggerMessageContent,
      }),
    );
    _checkError(response);
    return ContextAwareReply.fromJson(jsonDecode(response.body));
  }

  // ==================== 健康检查 ====================

  /// 健康检查
  Future<Map<String, dynamic>> healthCheck() async {
    final response = await _client.get(
      Uri.parse('$baseUrl/context-aware-reply/health'),
      headers: _headers,
    );
    _checkError(response);
    return jsonDecode(response.body) as Map<String, dynamic>;
  }

  void _checkError(http.Response response) {
    if (response.statusCode >= 400) {
      throw ApiException(
        statusCode: response.statusCode,
        message: response.body,
      );
    }
  }

  void dispose() {
    _client.close();
  }
}

/// API 异常
class ApiException implements Exception {
  final int statusCode;
  final String? message;

  ApiException({required this.statusCode, this.message});

  @override
  String toString() {
    return 'ApiException(statusCode: $statusCode, message: $message)';
  }
}

// 工具函数
extension ReplyIntentExtension on ReplyIntent {
  String get value => name;
  
  static ReplyIntent fromString(String value) {
    return ReplyIntent.values.firstWhere(
      (e) => e.name == value,
      orElse: () => ReplyIntent.SOCIAL,
    );
  }
  
  String get displayName {
    switch (this) {
      case ReplyIntent.BUSINESS:
        return '商务';
      case ReplyIntent.PERSONAL:
        return '个人';
      case ReplyIntent.SOCIAL:
        return '社交';
      case ReplyIntent.QUESTION:
        return '提问';
      case ReplyIntent.GREETING:
        return '问候';
      case ReplyIntent.APPRECIATION:
        return '感谢';
      case ReplyIntent.EMOTIONAL:
        return '情感';
    }
  }
}

extension LanguageStyleExtension on LanguageStyle {
  String get value => name;
  
  static LanguageStyle fromString(String value) {
    return LanguageStyle.values.firstWhere(
      (e) => e.name == value,
      orElse: () => LanguageStyle.CASUAL,
    );
  }
  
  String get displayName {
    switch (this) {
      case LanguageStyle.FORMAL:
        return '正式';
      case LanguageStyle.CASUAL:
        return '随意';
      case LanguageStyle.FRIENDLY:
        return '友好';
      case LanguageStyle.PROFESSIONAL:
        return '专业';
    }
  }
}

extension ReplyStatusExtension on ReplyStatus {
  String get value => name;
  
  static ReplyStatus fromString(String value) {
    return ReplyStatus.values.firstWhere(
      (e) => e.name == value,
      orElse: () => ReplyStatus.GENERATED,
    );
  }
  
  String get displayName {
    switch (this) {
      case ReplyStatus.GENERATED:
        return '已生成';
      case ReplyStatus.SELECTED:
        return '已选择';
      case ReplyStatus.REJECTED:
        return '已拒绝';
      case ReplyStatus.EXPIRED:
        return '已过期';
    }
  }
}